sdk/resource/resource.go

```diff
@@ -177,7 +178,16 @@ func Merge(a, b *Resource) (*Resource, error) {
 	case a.schemaURL == b.schemaURL:
 		schemaURL = a.schemaURL
 	default:
-		return Empty(), errMergeConflictSchemaURL
+		// Log conflict, but do not return an error. The user is not always
+		// able to handle these errors directly (i.e. resource detectors from
+		// 3rd-party packages). Returning something, and logging information,
+		// here is a better user-story than returning an unrecoverable error.
+		global.Error(
+			errMergeConflictSchemaURL,
+			"the merged resource schema URL is empty",
+			"URL 1", a.schemaURL,
+			"URL 2", b.schemaURL,
+		)
```

Comment:
How about `return merged, errMergeConflictSchemaURL`.

The doc comment can be changed to 

```go
// If the resources have different non-empty schemaURL a merged resource with empty scheme URL and an error
// will be returned.
```

Then the users can simply decide to ignore the error if they do not care about the conflict. For ones who care the error handling will be easier. I also think that this kind of change will result in more backwards-compatible behavior.

Reply:
As mentioned in the PR description. Many users who use resource detectors do not have access to this error. By returning the error here it will fail the detection entirely. Hence why the choice was made.

Reply:
Additionally, similar to our instrument name conflict error in the SDK, many users are confused if they can used the returned value when an error is returned. They often blindly just ...

```go
if err != nil {
    return
}
```

Given that is only going to continue user dissatisfaction with this project, I do not want to continue in that direction. Especially, since this is something the specification has clear language on regarding what resource should be returned.

Reply:
Hmm, looking at the detect code:

https://github.com/open-telemetry/opentelemetry-go/blob/d9d9507227b30ad83500ec59e5714d0314ac83be/sdk/resource/auto.go#L71-L74

It does seem like we continue even if there is an error actually. So the detector error would not stop the returning of a resource in that situation.

Reply:
You are right. You can consider adding `## Why not error and return merged?` to the PR description (it can help other reviewers). Feel free to resolve

Reply:
> If we went that route we would need to export the "recoverable" errors that can be treated as warnings and update the docs to make this clear.

I thought about exporting an error type e.g. `ErrResourceMergeConfict` it could even contain fields with scheme URLs from both sources.

Reply:
> I'm still not sure we want to introduce the confusion of returning a usable value and an error.

There are certain APIs which does it.

E.g. [`io.Reader`](https://pkg.go.dev/io#Reader) may return a result together with `io.EOF` error.

Although, I agree if does not feel idiomatic I think we should consider this approach.

I think we should also explore if/how we would like to add support resource transformations and how they will fit into the current design.

Reply:
I have little desire to add resource transformations until OTel has a stable implementation. Based on previous conversations that seems quite far off.

I expect we can follow what other SIGs do when they are supported. I don't think we need to address the issue here.

Reply:
>  I don't think we need to address the issue here.

I do not say we should address it. Just think if it would be a trouble if it is added. I do not expect it will be a problem 😉

Reply:
Updated.